tests/test_extract_declared_dependencies_pyproject_toml.py

```diff
@@ -176,118 +158,104 @@
         ),
     ],
 )
-def test_parse_pyproject_content__pep621_or_poetry_dependencies__yields_dependencies(
-    pyproject_toml, expected_deps
+def test_parse_pyproject_toml__pep621_or_poetry_dependencies__yields_dependencies(
+    write_tmp_files, pyproject_toml, expected_deps
 ):
-    source = Location(Path("pyproject.toml"))
-    result = list(parse_pyproject_contents(pyproject_toml, source))
-    expected = [DeclaredDependency(dep, source) for dep in expected_deps]
+    tmp_path = write_tmp_files({"pyproject.toml": pyproject_toml})
+    path = tmp_path / "pyproject.toml"
+
+    result = list(parse_pyproject_toml(path))
+    expected = [DeclaredDependency(dep, Location(path)) for dep in expected_deps]
```

Comment:
The lists are compared for full equivalence rather than just contents; why does order matter here?